### ISSUE TITLE
Create DataSet classes loading Training, Validation and Test Images.

### ISSUE THREAD

**Issue by koushikvikram** [koushikvikram/multimodal-image-retrieval#11]
For training and validation sets, load the caption embeddings and corresponding images in memory.
For test set, load only images in memory. Don't load captions.

We'll use images along with their caption embeddings to train an "encoder" that learns how to embed images in caption embedding space.

**Comment by koushikvikram** [comment]
Created CaptionEmbeddingsDataset and TestImagesDataset classes in [src/dataset.py](https://github.com/koushikvikram/multimodal-image-retrieval/blob/c144f22722f16ce63cd81d5fdfc9362c1d28600a/src/dataset.py)
